### ISSUE TITLE
Check the package and the website doesn't work - [platform: Win10 x64]

### ISSUE THREAD

**Issue by ahmetislam25** [wbond/package_control#1045]
The website does not open

![capture](https://cloud.githubusercontent.com/assets/9280397/12071858/43ef92c2-b0cb-11e5-811e-b0890ea85872.JPG)

Installed the Package Control , but this error occur while I'm trying to install package

![3e361194-aa90-11e5-8e2d-7ea880f303d0](https://cloud.githubusercontent.com/assets/9280397/12071865/6a5c5350-b0cb-11e5-9658-a9d701691d54.png)

Package control is not installed

```
warning: mnemonic t not found in menu caption Tools
startup time: 0.144587
first paint time: 0.144587
reloading plugin Default.block
reloading plugin Default.comment
reloading plugin Default.copy_path
reloading plugin Default.delete_word
reloading plugin Default.detect_indentation
reloading plugin Default.duplicate_line
reloading plugin Default.echo
reloading plugin Default.exec
reloading plugin Default.fold
reloading plugin Default.font
reloading plugin Default.goto_line
reloading plugin Default.history_list
reloading plugin Default.indentation
reloading plugin Default.kill_ring
reloading plugin Default.mark
reloading plugin Default.new_templates
reloading plugin Default.open_context_url
reloading plugin Default.open_file_settings
reloading plugin Default.open_in_browser
reloading plugin Default.pane
reloading plugin Default.paragraph
reloading plugin Default.paste_from_history
reloading plugin Default.quick_panel
reloading plugin Default.save_on_focus_lost
reloading plugin Default.scroll
reloading plugin Default.set_unsaved_view_name
reloading plugin Default.side_bar
reloading plugin Default.sort
reloading plugin Default.swap_line
reloading plugin Default.switch_file
reloading plugin Default.symbol
reloading plugin Default.transform
reloading plugin Default.transpose
reloading plugin Default.trim_trailing_white_space
reloading plugin CSS.css_completions
reloading plugin Diff.diff
reloading plugin HTML.encode_html_entities
reloading plugin HTML.html_completions
reloading plugin 0_package_control_loader.00-package_control
Package Control: Error finding main directory from loader
reloading plugin 0_package_control_loader.01-pygments
Traceback (most recent call last):
  File "C:\Program Files\Sublime Text 3\sublime_plugin.py", line 74, in reload_plugin
    m = importlib.import_module(modulename)
  File "./importlib/__init__.py", line 90, in import_module
  File "<frozen importlib._bootstrap>", line 1584, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "C:\Program Files\Sublime Text 3\sublime_plugin.py", line 678, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "01-pygments in C:\Users\Ahmet İslam\AppData\Roaming\Sublime Text 3\Installed Packages\0_package_control_loader.sublime-package", line 1, in <module>
ImportError: No module named 'package_control'
reloading plugin 0_package_control_loader.02-bz2
Traceback (most recent call last):
  File "C:\Program Files\Sublime Text 3\sublime_plugin.py", line 74, in reload_plugin
    m = importlib.import_module(modulename)
  File "./importlib/__init__.py", line 90, in import_module
  File "<frozen importlib._bootstrap>", line 1584, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "C:\Program Files\Sublime Text 3\sublime_plugin.py", line 678, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "02-bz2 in C:\Users\Ahmet İslam\AppData\Roaming\Sublime Text 3\Installed Packages\0_package_control_loader.sublime-package", line 1, in <module>
ImportError: No module named 'package_control'
reloading plugin 0_package_control_loader.50-backrefs
Traceback (most recent call last):
  File "C:\Program Files\Sublime Text 3\sublime_plugin.py", line 74, in reload_plugin
    m = importlib.import_module(modulename)
  File "./importlib/__init__.py", line 90, in import_module
  File "<frozen importlib._bootstrap>", line 1584, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "C:\Program Files\Sublime Text 3\sublime_plugin.py", line 678, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "50-backrefs in C:\Users\Ahmet İslam\AppData\Roaming\Sublime Text 3\Installed Packages\0_package_control_loader.sublime-package", line 1, in <module>
ImportError: No module named 'package_control'
reloading plugin Alignment.Alignment
reloading plugin All Autocomplete.all_views_completions
reloading plugin AutoFileName.autofilename
reloading plugin AutoFileName.getimageinfo
reloading plugin BracketHighlighter.bh_core
Traceback (most recent call last):
  File "C:\Program Files\Sublime Text 3\sublime_plugin.py", line 74, in reload_plugin
    m = importlib.import_module(modulename)
  File "./importlib/__init__.py", line 90, in import_module
  File "<frozen importlib._bootstrap>", line 1584, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "C:\Program Files\Sublime Text 3\sublime_plugin.py", line 678, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "bh_core in C:\Users\Ahmet İslam\AppData\Roaming\Sublime Text 3\Installed Packages\BracketHighlighter.sublime-package", line 16, in <module>
  File "C:\Program Files\Sublime Text 3\sublime_plugin.py", line 678, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "bh_rules in C:\Users\Ahmet İslam\AppData\Roaming\Sublime Text 3\Installed Packages\BracketHighlighter.sublime-package", line 7, in <module>
ImportError: No module named 'backrefs'
reloading plugin BracketHighlighter.bh_logging
reloading plugin BracketHighlighter.bh_plugin
reloading plugin BracketHighlighter.bh_regions
reloading plugin BracketHighlighter.bh_remove
reloading plugin BracketHighlighter.bh_rules
Traceback (most recent call last):
  File "C:\Program Files\Sublime Text 3\sublime_plugin.py", line 72, in reload_plugin
    m = imp.reload(m)
  File "./imp.py", line 276, in reload
  File "C:\Program Files\Sublime Text 3\sublime_plugin.py", line 678, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "bh_rules in C:\Users\Ahmet İslam\AppData\Roaming\Sublime Text 3\Installed Packages\BracketHighlighter.sublime-package", line 7, in <module>
ImportError: No module named 'backrefs'
reloading plugin BracketHighlighter.bh_search
reloading plugin BracketHighlighter.bh_swapping
reloading plugin BracketHighlighter.bh_wrapping
reloading plugin DocBlockr.jsdocs
reloading plugin Emmet.emmet-plugin
reloading plugin HTML5.encode_html_entities
reloading plugin HTML5.html_completions
reloading plugin JsFormat.js_formatter
reloading plugin LESS.less_completions
reloading plugin Markdown HTML Preview.MarkdownHtmlPreview
reloading plugin MarkdownEditing.bootstrap
reloading plugin MarkdownEditing.custom_find_under_expand
reloading plugin MarkdownEditing.decide_title
reloading plugin MarkdownEditing.distraction_free_mode
reloading plugin MarkdownEditing.footnotes
reloading plugin MarkdownEditing.gather_missing_links
reloading plugin MarkdownEditing.indent_list_item
reloading plugin MarkdownEditing.indent_list_multiitem
reloading plugin MarkdownEditing.insert_references
reloading plugin MarkdownEditing.jumpToMarker
reloading plugin MarkdownEditing.lint
reloading plugin MarkdownEditing.list_markdown_references
reloading plugin MarkdownEditing.numbered_list
reloading plugin MarkdownEditing.paste_as_link
reloading plugin MarkdownEditing.paste_as_reference
reloading plugin MarkdownEditing.quote_indenting
reloading plugin MarkdownEditing.switch_list_bullet_type
reloading plugin MarkdownEditing.underlined_headers
reloading plugin Nodejs.__init__
reloading plugin Nodejs.Nodejs
reloading plugin SassBeautify.SassBeautify
reloading plugin SCSS.scss_completions
reloading plugin SideBarEnhancements.SideBar
reloading plugin SideBarEnhancements.SideBarAPI
reloading plugin SideBarEnhancements.SideBarDefaultDisable
reloading plugin SideBarFolders.SideBarFolders
plugins loaded
warning: mnemonic t not found in menu caption Tools
Emmet: No need to update PyV8
>>> import urllib.request,os,hashlib; h = '2915d1851351e5ee549c20394736b442' + '8bc59f460fa1548d1514676163dafc88'; pf = 'Package Control.sublime-package'; ipp = sublime.installed_packages_path(); urllib.request.install_opener( urllib.request.build_opener( urllib.request.ProxyHandler()) ); by = urllib.request.urlopen( 'http://packagecontrol.io/' + pf.replace(' ', '%20')).read(); dh = hashlib.sha256(by).hexdigest(); print('Error validating download (got %s instead of %s), please try manual install' % (dh, h)) if dh != h else open(os.path.join( ipp, pf), 'wb' ).write(by)
Traceback (most recent call last):
  File "./urllib/request.py", line 1248, in do_open
  File "./http/client.py", line 1065, in request
  File "./http/client.py", line 1103, in _send_request
  File "./http/client.py", line 1061, in endheaders
  File "./http/client.py", line 906, in _send_output
  File "./http/client.py", line 844, in send
  File "./http/client.py", line 822, in connect
  File "./socket.py", line 435, in create_connection
  File "./socket.py", line 426, in create_connection
TimeoutError: [WinError 10060] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "./urllib/request.py", line 156, in urlopen
  File "./urllib/request.py", line 469, in open
  File "./urllib/request.py", line 487, in _open
  File "./urllib/request.py", line 447, in _call_chain
  File "./urllib/request.py", line 1268, in http_open
  File "./urllib/request.py", line 1251, in do_open
urllib.error.URLError: <urlopen error [WinError 10060] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond>

```

**Comment by karlhorky** [comment]
I'm also getting this on OSX. It appears packagecontrol.io is down?

**Comment by DaFri-Nochiterov** [comment]
Site is down.
![image](https://cloud.githubusercontent.com/assets/10401817/12072217/87a9e9d0-b0fe-11e5-9dbf-664fbc42ef43.png)

**Comment by Marcel-Robitaille** [comment]
Same here. Hope it gets fixed soon!

**Comment by ahmetislam25** [comment]
see their tweets on Twitter

**Comment by Marcel-Robitaille** [comment]
@ahmetislam25 Who's twitter?

**Comment by karlhorky** [comment]
I think Ahmet is referring to @wbond's Twitter: https://twitter.com/wbond/status/683059802533195776

**Comment by karlhorky** [comment]
@wbond is there any way to work around these issues until connectivity is restored? Maybe saving a local version of the channel json in package control's directory and forcing it to use this cache?

**Comment by wbond** [comment]
@karlhorkey Caching automatically happens, but the default lifetime is 5 minutes, and it is kept in memory, so it resets when ST is restarted.

Additionally, HTTP requests are cached on disk, but currently are only used to allow servers to respond with a 304 when a resource has not changed.

Additional logic would be required to try and figure out when a connection-level failure occurred and whether a cached response was recent enough to use.

**Comment by ahmetislam25** [comment]
opened website a few minutes ago. 5 minutes later went back to the site.

**Comment by karlhorky** [comment]
@wbond ah ok so could a workaround be possible like this:

- run a local server that responds with 304s for the channel json file
- add an entry to the hosts file to point packagecontrol.io at this local server
- change the date on the local cache to be less than 5 minutes old

I still have a copy of the json cached from December 27, so this will probably be sufficient to install some packages. Would this work?

Maybe for the future, a fallback server can be considered. Maybe it can be hosted from GitHub Pages?

**Comment by FichteFoll** [comment]
Although it is just json, the server still has to deliver a lot of data. I'm unsure if "abusing" github pages is the right call here. 

To me, it sounds like the best idea atm to increase the caching duration to a few days and fall back to it in case of connection failure.

**Comment by wbond** [comment]
Package Control moves between hundreds of gigabytes to over a terabyte of JSON data per day. The only way it is possible without massive bandwidth bills is a custom bz2 transfer encoding. Putting raw JSON anywhere would be an abuse of any sort of free hosting.

I could spend a bunch of my time, unpaid, to try to find inexpensive or free ways that users can continue to use Package Control in the very rare even of Linode issues. However, I feel trying to build a sustainable funding of Package Control from the hundreds of thousands of daily users, so that extra infrastructure can be purchased, is a better plan.

**Comment by karlhorky** [comment]
Hm, yeah the data volume sounds like it's definitely an issue. 

How feasible would it be that users could just clone the packagecontrol.io repo from GitHub and start a local server? Not sure if the repo just works out of the box like that. Still wouldn't be an effortless, "it just works" fallback, but maybe it could be a stopgap workaround until there's more funding for infrastructure.

**Comment by karlhorky** [comment]
If this is not an option then maybe the easiest solution for now is what @FichteFoll mentioned.

**Comment by TheAwesomeGem** [comment]
I can't access it either. It's still down. :/

**Comment by ahmetislam25** [comment]
@TheAwesomeGem Yes, the problem continues :/

**Comment by SupremeTechnopriest** [comment]
@wbond Thanks for your awesome work on this project.  I totally agree that we need a community sponsored hardware failover.  How much would we be looking at to set up an instance in another datacenter... say newark? I would be willing to donate to see this happen.

**Comment by Hualin** [comment]
please get it online, I will donate, please!

**Comment by ashuforweb** [comment]
me too, please get it online somehow

**Comment by DaFri-Nochiterov** [comment]
Site is online now

wew!

**Comment by TheShotte** [comment]
Here I was freaking out thinking something is wrong with my install (ST2 says I'm missing dependencies which are clearly there). Trying to get the SFTP plugin for ST2 for which I have a paid license.

**Comment by karlhorky** [comment]
The Internet Archive cached the channel JSON in September 2015: 
https://web.archive.org/web/20150905194312/https://packagecontrol.io/channel_v3.json

Temporarily changing your package control settings to use this URL as a channel allows installation of packages again: https://github.com/wbond/package_control/issues/959#issuecomment-168447848

**Comment by DaFri-Nochiterov** [comment]
@karlhorky >_<
![image](https://cloud.githubusercontent.com/assets/10401817/12078886/7a9f8c40-b24d-11e5-84e9-3fc1736bc042.png)

**Comment by karlhorky** [comment]
@DaFri-Nochiterov Where are you getting this error? Both links work for me...

**Comment by tobiajo** [comment]
My Sublime Text application is useless as long as this continues. Is there any alternatives to Package Control?

**Comment by limzykenneth** [comment]
https://packagecontrol.io is still down and I'm still having this message:
![screen shot 2016-01-03 at 1 18 00 pm](https://cloud.githubusercontent.com/assets/7543950/12078921/751932f2-b21c-11e5-8e3e-cff672f30464.png)
Guess still not up huh?

**Comment by maciejgluszek** [comment]
@karlhorky So when the main feed comes back, all you need to do is remove the archive.org entry and it will update itself and packages if neded? Thanks.

**Comment by dharma-guardian** [comment]
@tobiajo You can always install packages manually. So, "useless" is a bit exaggerated.

**Comment by karlhorky** [comment]
@maciejgluszek Yeah as far as I know, once the service is up again you can just remove this setting from your user config again and everything will update to the new versions.

**Comment by mohjak** [comment]
@karlhorky Thanks a lot this URL works for me :+1:

**Comment by DaFri-Nochiterov** [comment]
@karlhorky, archive.org not works in Russia, very strange. I used proxy and all works fine.
Ooh, Package Control also can't load this file.

```
Package Control: Error downloading channel. URL error [Errno 104] Connection reset by peer downloading https://web.archive.org/web/20150905194312/https://packagecontrol.io/channel_v3.json.
```

Can I load local file?

**Comment by oxapentane** [comment]
@DaFri-Nochiterov several archive.org subdomains are blocked in Russia. Send your love to Russian government.

**Comment by DaFri-Nochiterov** [comment]
@oxapentane I don't want to send my love to Russian government. My love for other purposes ;)

**Comment by ahmetislam25** [comment]
@mohjak how to use it?

**Comment by mohjak** [comment]
As @wbond refers instructions available at the following #1057
